src/appliance_setup/avs/avsarconboarder/retriever/vsphere/vsphere_details_retriever.py

```diff
@@ -1,24 +1,104 @@
+        datastore_id, err = govc_cli('find', '-i=true', 'datastore', '-name', data_store_name)
+        if err:
+            raise vCenterOperationFailed('Retrieval of datastore managed object failed')
+        logging.info("Retrieved datastore managed object successfully")
+        res, err = govc_cli('find', '.', '-type', 'c', '-datastore', datastore_id)
+        if err:
+            raise vCenterOperationFailed('Retrieving of associated clusters failed')
+        logging.info("Retrieved associated clusters successfully")
+        clusters = res.split("\n")
+        return clusters
+    
+    def _retrieve_data_stores(self):
+        res, err = govc_cli('find', '-type', 's', ' -json=true')
+        res = json.loads(res)
+        if err:
+            raise vCenterOperationFailed('Retrieving of Datastores failed')
+        logging.info("Datastores info retrieved successfully")
+        return res
+    
+    def _set_vcenter_cred(self, cloud_details, vcenter_credentials):
+        url_data = urlparse(cloud_details['vcsa_endpoint'])
+        vCenter_address= url_data.netloc + ':443'
+        os.environ['GOVC_INSECURE'] = "true"
+        os.environ['GOVC_URL'] = f"https://{vCenter_address}/sdk"
+        os.environ['GOVC_USERNAME'] = f"{vcenter_credentials.username}"
+        os.environ['GOVC_PASSWORD'] = f"{vcenter_credentials.password}"
+    
+    def _retrieve_default_values_if_present(self, data_stores):   
+        cluster_path = None
+        data_store_name = None   
+        for data_store in data_stores:
+            if data_store is not None and data_store != '':
+                data_store_path_seperated = data_store.split("/")
+                if(data_store_path_seperated[-1] == "vsanDatastore"):
+                    data_store_name = data_store_path_seperated[-1]
+                    clusters = self._find_clusters(data_store_name)
+                    for cluster in clusters:
+                        if cluster is not None and cluster != '':
+                            cluster_path_seperated = cluster.split("/")
+                            if(cluster_path_seperated[-1] == "Cluster-1"):
+                                cluster_path = cluster
+
+        return data_store_name, cluster_path
+
+    def _retrieve_environment_details(self):
+        data_stores = self._retrieve_data_stores()
+
+        try:
+            default_data_store, default_cluster_path = self._retrieve_default_values_if_present(data_stores)
+            if default_data_store != None and default_cluster_path != None:
+                logging.info("Default DataStore, Cluster available")
+                return default_data_store, default_cluster_path
+            else:
+                logging.info("Default DataStore, Cluster Not Available")
+        except:
+            logging.info("Error in fetching default DataStore, Cluster Not Available")
+
+        cluster_path = None
+        data_store_name = None   
+        for data_store in data_stores:
```

Comment:
NIT: There is no cluster level backward compatibility. Can you add that?

Reply:
Let's target this in next PR. We need backward compatibility only in offboard experience and this is not exposed to customer for now. We always have an option to provide these values as input in config though.

Reply:
https://msazure.visualstudio.com/One/_workitems/edit/25758005 - created a task for this in backlog